### ISSUE TITLE
omnisharp and company-mode fail to complete 

### ISSUE THREAD

**Issue by tattsun** [OmniSharp/omnisharp-emacs#153]
Hello.

I installed omnisharp, omnisharp-emacs and company-mode. And I have an issue about completion.

* GNU Emacs 24.4.1 on mac(yosemite)
* omnisharp cebdc29cf945e460bf074d48d2c9acab2c2eaf03
* omnisharp-emacs 3358df8abec334425db2b1d67488aecc2af192e9
* company-mode af962d47817fb232f23f527e4ee1bcb61f31bbcf

When I try to use company-mode's completion, I encounter the error below.
How can I fix it?

```
Debugger entered--Lisp error: (wrong-number-of-arguments #[(var) "\301\302\303DDE\207" [var and boundp quote] 4 1795186] 3)
  #[(var) "\301\302\303DDE\207" [var and boundp quote] 4 1795186](omnisharp-mode (not (company-in-string-or-comment)) (omnisharp-company--prefix))
  (bound-and-true-p omnisharp-mode (not (company-in-string-or-comment)) (omnisharp-company--prefix))
  (cond ((eql command (quote prefix)) (bound-and-true-p omnisharp-mode (not (company-in-string-or-comment)) (omnisharp-company--prefix))) ((eql command (quote candidates)) (omnisharp--get-company-candidates arg)) ((eql command (quote no-cache)) (equal arg "")) ((eql command (quote annotation)) (omnisharp--company-annotation arg)) ((eql command (quote meta)) (omnisharp--get-company-candidate-data arg (quote DisplayText))) ((eql command (quote require-match)) (quote never)) ((eql command (quote doc-buffer)) (let ((doc-buffer (company-doc-buffer (omnisharp--get-company-candidate-data arg (quote Description))))) (with-current-buffer doc-buffer (visual-line-mode)) doc-buffer)) ((eql command (quote ignore-case)) omnisharp-company-ignore-case) ((eql command (quote sorted)) omnisharp-company-sort-results) ((eql command (quote post-completion)) (let* ((json-result (get-text-property 0 (quote omnisharp-item) arg)) (allow-templating (get-text-property 0 (quote omnisharp-allow-templating) arg))) (omnisharp--tag-text-with-completion-info arg json-result) (when allow-templating (if (and omnisharp-company-template-use-yasnippet (fboundp (quote yas/expand-snippet))) (progn (let (...) (when method-snippet ...))) (require (quote company-template)) (let ((method-base ...)) (when (and method-base ...) (company-template-c-like-templatify method-base))))))))
  (cl-case command (prefix (bound-and-true-p omnisharp-mode (not (company-in-string-or-comment)) (omnisharp-company--prefix))) (candidates (omnisharp--get-company-candidates arg)) (no-cache (equal arg "")) (annotation (omnisharp--company-annotation arg)) (meta (omnisharp--get-company-candidate-data arg (quote DisplayText))) (require-match (quote never)) (doc-buffer (let ((doc-buffer (company-doc-buffer (omnisharp--get-company-candidate-data arg (quote Description))))) (with-current-buffer doc-buffer (visual-line-mode)) doc-buffer)) (ignore-case omnisharp-company-ignore-case) (sorted omnisharp-company-sort-results) (post-completion (let* ((json-result (get-text-property 0 (quote omnisharp-item) arg)) (allow-templating (get-text-property 0 (quote omnisharp-allow-templating) arg))) (omnisharp--tag-text-with-completion-info arg json-result) (when allow-templating (if (and omnisharp-company-template-use-yasnippet (fboundp (quote yas/expand-snippet))) (progn (let (...) (when method-snippet ...))) (require (quote company-template)) (let ((method-base ...)) (when (and method-base ...) (company-template-c-like-templatify method-base))))))))
  company-omnisharp(prefix)
  #[0 "\n\203	\nC\202\n\211\205\220\211@\3009\204\306!\203<9\2036\307N\310=\2046\307N?\205@\311!\205@\312!\202@\313\312\"\240\210\300\242\203\211\314\300\242!\203\203\315\300\242!\211\301\316\f!\240\210\301\242:\204l
\203\203\317\320!\210\202\203
\203t\300\242\321\301\242!\210\322\323\324 \"\210\325\326!\210\327\330\301\242\"\210A\266\202\202\n\207" [(nil) (nil) company-backend company-backends company-prefix company--manual-action functionp company-init t company-init-backend prefix company--multi-backend-adapter company--good-prefix-p company--prefix-str company-calculate-candidates message "No completion found" company-update-candidates run-hook-with-args company-completion-started-hook company-explicit-action-p company-call-frontends show throw --cl-block-nil-- company--manual-prefix] 6 "\n\n(fn)"]()
  funcall(#[0 "\n\203	\nC\202\n\211\205\220\211@\3009\204\306!\203<9\2036\307N\310=\2046\307N?\205@\311!\205@\312!\202@\313\312\"\240\210\300\242\203\211\314\300\242!\203\203\315\300\242!\211\301\316\f!\240\210\301\242:\204l
\203\203\317\320!\210\202\203
\203t\300\242\321\301\242!\210\322\323\324 \"\210\325\326!\210\327\330\301\242\"\210A\266\202\202\n\207" [(nil) (nil) company-backend company-backends company-prefix company--manual-action functionp company-init t company-init-backend prefix company--multi-backend-adapter company--good-prefix-p company--prefix-str company-calculate-candidates message "No completion found" company-update-candidates run-hook-with-args company-completion-started-hook company-explicit-action-p company-call-frontends show throw --cl-block-nil-- company--manual-prefix] 6 "\n\n(fn)"])
  company--begin-new()
  company--perform()
  #[0 "\301 \210\207" [company-candidates company--perform] 1 "\n\n(fn)"]()
  funcall(#[0 "\301 \210\207" [company-candidates company--perform] 1 "\n\n(fn)"])
  company-auto-begin()
  company-idle-begin(#<buffer MTPClient.cs> #<window 3 on MTPClient.cs> 180 157)
  apply(company-idle-begin (#<buffer MTPClient.cs> #<window 3 on MTPClient.cs> 180 157))
  byte-code("r\301\302H\303H\"\210)\301\207" [timer apply 5 6] 4)
  timer-event-handler([t 21748 44389 937269 nil company-idle-begin (#<buffer MTPClient.cs> #<window 3 on MTPClient.cs> 180 157) nil 0])
```

**Comment by sp3ctum** [comment]
@nosami @bbbscarter do you get this too? What could be wrong? I don't use
company mode personally. At least not yet. :)
2.3.2015 20.43 kirjoitti "tattsun" <notifications@github.com>:

**Comment by nosami** [comment]
Looks like the issue that was fixed here https://github.com/bbbscarter/omnisharp-emacs/commit/f1ae6000887af7816911c9a2ad4c3394a726c172

It's part of https://github.com/OmniSharp/omnisharp-emacs/pull/151

**Comment by bbbscarter** [comment]
Yup - although the fix was slightly wrong. I’ve properly fixed it in my repo, but it hasn’t been pulled upstream yet.



On 3 March 2015 at 06:48:54, Jason Imison (notifications@github.com) wrote:

Looks like the issue that was fixed here bbbscarter@f1ae600

It's part of #151

**Comment by tattsun** [comment]
I've patched my code according to bbbscarter@f1ae600, then it's fixed!

Thanks a lot:smile:
